### PR TITLE
Stats: Improvements to video stats

### PR DESCRIPTION
**Pull request by thedebian**
#### Changes proposed in this Pull Request

This PR updates video stats to add watch time and impressions.
It also changes "plays" to "views" and fixes some existing issues with Video stats.

#### Testing instructions

* On a sanboxed public-api
* Apply D79611-code
* Apply this PR
* Open your Calypso backend
* Select the "Stats" menu
* ✅ The "video stats" section should work.
* Click on the video stats
* ✅ You should see a list of video stats for the selected period
* ✅ You should be able to click on "Download CSV" and generate a CSV from the visible stats
* On a stat line, click on the video title
* ✅ You should be redirected to the media item in the Media library
* Go back
* Click on each stats (views, impressions and hours watched)
* ✅ Each stat should redirect to the appropriate stat section (with some bars)
* ✅ The stats should match
* ✅ The stats should reflect your currently selected time period
* ✅ Using the "year" time period, the graph should display month-by-month
* ✅ Using another time perios, the graph should be daily

Related to Automattic/greenhouse#1170

### CODE REVIEW

**Review by thedebian — COMMENTED** [review]
Code review seems 👍

File: client/my-sites/stats/stats-strings.js [hunk 1]
```diff
@@ -72,10 +72,10 @@ export default function () {
 	statsStrings.videoplays = {
 		title: translate( 'Videos', { context: 'Stats: title of module' } ),
 		item: translate( 'Video', { context: 'Stats: module row header for videos.' } ),
-		value: translate( 'Plays', {
-			context: 'Stats: module row header for number of plays per video.',
+		value: translate( 'Views', {
+			context: 'Stats: module row header for number of views per video.',
 		} ),
-		empty: translate( 'No videos played', {
+		empty: translate( 'No videos views', {
```
Comment (jgcaruso) [inline client/my-sites/stats/stats-strings.js]:
```suggestion
		empty: translate( 'No videos viewed', {
```

Or `No video views`

File: client/my-sites/stats/videopress-stats-module/index.jsx [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+			completeVideoStats = Object.values( data.days )
+				.map( ( o ) => o.data )
+				.flat();
+		}
+
+		const noData = data && this.state.loaded && ! completeVideoStats.length;
+		// Only show loading indicators when nothing is in state tree, and request in-flight
+		const isLoading = ! this.state.loaded && ! ( data && data.length );
+
+		// TODO: Support error state in redux store
```
Comment (jgcaruso) [inline client/my-sites/stats/videopress-stats-module/index.jsx]: Errant `TODO` - not sure we planned to do this or it was copy-pastad?

Comment (roundhill) [reply client/my-sites/stats/videopress-stats-module/index.jsx]: Probably 🍝 ?

Comment (roundhill) [reply client/my-sites/stats/videopress-stats-module/index.jsx]: Verified that it was copied, removed in https://github.com/Automattic/wp-calypso/pull/63173/commits/5ef4172f29c6323a4c9e91647f9b48bac07a02dc

**Review by jgcaruso — APPROVED** [review]
Looks good!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7279210

Hi @thedebian, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Hours Watched`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7279210).

**Pull request merged**